var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -25,29 +25,47 @@
 from spack import *
 
 
-class Zfp(Package):
+class Zfp(MakefilePackage):
     """zfp is an open source C library for compressed floating-point arrays
        that supports very high throughput read and write random acces,
        target error bounds or bit rates.  Although bit-for-bit lossless
        compression is not always possible, zfp is usually accurate to
        within machine epsilon in near-lossless mode, and is often orders
-       of magnitude more accurate than other lossy compressors.
-
+       of magnitude more accurate than other lossy compressors. Versions
+       of zfp 0.5.1 or newer also support compression of integer data.
     """
 
     homepage = "http://computation.llnl.gov/projects/floating-point-compression"
-    url      = "http://computation.llnl.gov/projects/floating-point-compression/download/zfp-0.5.0.tar.gz"
+    url      = "http://computation.llnl.gov/projects/floating-point-compression/download/zfp-0.5.1.tar.gz"
 
+    version('0.5.1', '0ed7059a9b480635e0dd33745e213d17')
     version('0.5.0', '2ab29a852e65ad85aae38925c5003654')
 
+    variant('bswtuint8', default=False,
+            description='Build with bit stream word type of uint8')
+
+    def edit(self, spec, prefix):
+        if '+bswtuint8' in self.spec:
+            config_file = FileFilter('Config')
+            config_file.filter(
+                '^\s*#\s*DEFS\s*\+=\s*-DBIT_STREAM_WORD_TYPE\s*=\s*uint8',
+                'DEFS += -DBIT_STREAM_WORD_TYPE=uint8')
+
     def install(self, spec, prefix):
```

Comment:
You'll want to split the install phase into two phases, `build` and `install`. The way things are now, Spack runs the default `MakefilePackage` `build` phase which just runs `make`. You want:
```python
def build(self, spec, prefix):
    make('shared')

def install(self, spec, prefix):
    zfp_incdir ...
```
Alternatively, you could add:
```python
build_targets = ['shared']
```
to the top of the package and leave out the `build` phase.